offchain-modules/lib/src/transfer/to_ckb.rs

```diff
@@ -24,12 +23,15 @@ pub fn approve(from: H160, to: H160, url: String, chain_id: u32, key_path: Strin
         }],
         constant: false,
     };
-    let data = function_encode(function);
-    rpc_client.send_transaction(from, to, key_path, data)
+    let tokens = [Token::Address(from), Token::Uint(U256::max_value())];
+    let input_data = function.encode_input(&tokens).expect("invalid input data");
+    let f = rpc_client.send_transaction(from, to, key_path, input_data, U256::from(0));
+    let mut rt = Runtime::new().unwrap();
+    rt.block_on(f).expect("invalid tx hash")
```

Comment:
Create a runtime in every function is a bad practice.
You can add async for all the functions all the way up to `main` and create the runtime there(you can use the macro to make it easier).

```rust
#[tokio::main]
async fn main() {
    let v = example().await
    println!("{}", v);
}
```

Or turn the async in libraries to sync in the very beginning. In our case, it is `eth_util.rs`.

For example, `get_block_with_number` can be rewritten as below.

```rust
    pub fn get_block_with_number(&mut self, number: usize) -> (Vec<u8>, H256) {
        let block_header = futures::executor::block_on(
            self.client
                .eth()
                .block(BlockId::Number(BlockNumber::Number((number as u64).into()))),
        );
        let header = block_header.expect("invalid header");
        let mut stream = RlpStream::new();
        rlp_append(&header.clone().unwrap(), &mut stream);
        let header_vec = stream.out();
        println!("header rlp: {:?}", hex::encode(header_vec.clone()));
        (header_vec, H256(header.unwrap().hash.unwrap().0))
    }
```

ref: https://stackoverflow.com/questions/52521201/how-do-i-synchronously-return-a-value-calculated-in-an-asynchronous-future-in-st